.github/workflows/linux.yml

```diff
@@ -1,51 +1,46 @@
 name: Linux builds
 on: push
-jobs:
+env:
+  MAX_WARNINGS_GCC_6_Debug:     378
+  MAX_WARNINGS_GCC_6_Release:   661
+  MAX_WARNINGS_GCC_7_Debug:     380
+  MAX_WARNINGS_GCC_7_Release:   329
+  MAX_WARNINGS_GCC_8_Debug:     380
+  MAX_WARNINGS_GCC_8_Release:   330
+  MAX_WARNINGS_GCC_9_Debug:     403
+  MAX_WARNINGS_GCC_9_Release:   358
+  MAX_WARNINGS_Clang_7_Debug:   193
+  MAX_WARNINGS_Clang_7_Release: 193
+  MAX_WARNINGS_Clang_8_Debug:   193
+  MAX_WARNINGS_Clang_8_Release: 193
 
-  build_ubuntu_clang_8:
-    name: Clang 8 (ubuntu-18.04)
-    runs-on: ubuntu-18.04
-    steps:
-      - uses: actions/checkout@v1
-      - name: Log environment
-        run:  ./scripts/log-env.sh
-      - run:  sudo apt-get update
-      - name: Install dependencies
-        run:  sudo apt-get install -y $(./scripts/list-build-dependencies.sh -p apt --compiler clang --compiler-version 8)
-      - name: Build
-        run:  ./scripts/build.sh --compiler clang --compiler-version 8 --lto
-      - name: Summarize warnings
-        run:  ./scripts/count-warnings.py build.log
+jobs:
 
-  build_ubuntu_matrix_gcc:
-    name: GCC
-    runs-on: ${{ matrix.os }}
+  build_matrix:
+    name: ${{ matrix.compiler }}-${{ matrix.compiler_version }}
+    runs-on: ubuntu-latest
     strategy:
       matrix:
-        os: [ubuntu-18.04, ubuntu-16.04]
+        compiler: [Clang, GCC]
+        compiler_version: [6, 7, 8, 9]
+        # build_type: [Debug, Release]
+        exclude:
+          - compiler: Clang
+            compiler_version: 6
+          - compiler: Clang
+            compiler_version: 9
     steps:
       - uses: actions/checkout@v1
-      - name: Log environment
-        run:  ./scripts/log-env.sh
       - run:  sudo apt-get update
-      - name: Install dependencies
-        run:  sudo apt-get install -y $(./scripts/list-build-dependencies.sh -p apt)
-      - name: Build
-        run:  ./scripts/build.sh
-      - name: Summarize warnings
-        run:  ./scripts/count-warnings.py build.log
-
-  build_ubuntu_gcc9:
-    name: GCC 9 (ubuntu-18.04)
-    runs-on: ubuntu-18.04
-    steps:
-      - uses: actions/checkout@v1
+      - name: Install C++ compiler and libraries
+        run:  sudo apt-get install -y $(./scripts/list-build-dependencies.sh -m apt -c ${{ matrix.compiler }} -v ${{ matrix.compiler_version }})
       - name: Log environment
         run:  ./scripts/log-env.sh
-      - run:  sudo apt-get update
-      - name: Install dependencies
-        run:  sudo apt-get install -y $(./scripts/list-build-dependencies.sh -p apt --compiler-version 9)
-      - name: Build
-        run:  ./scripts/build.sh --compiler-version 9 --lto
-      - name: Summarize warnings
-        run:  ./scripts/count-warnings.py build.log
+      - name: Release build
+        run:  ./scripts/build.sh --compiler ${{ matrix.compiler }} --version-postfix ${{ matrix.compiler_version }} --build-type Release
```

Comment:
I don't want to use `build.sh` script for release builds… When trying to create a test release package, I discovered there are several steps missing from buildsystem (like the packaging of documentation, icons, translations, licenses) and I would prefer to have it explicitly listed in CI instead of being hidden in a script.

Release will need to be a separate job, as a failure in packaging should be explicitly visible in CI.

Also, I strongly prefer Release to be a job done in one configuration only (for Linux: Ubuntu 16.04 with default GCC) - this will make it easier to share builds with users/testers and gather feedback.

Reply:
Sure; basically a 100% explicit build where everything i spelled out step by step, including cflags exports and each command leading to the final package.  Makes sense!

Is that something you want to add, later, or should we add it to this PR?

Reply:
No, not as part of this PR; I was working offline on #31 (I was not assigned to it, oops!). I will make a set of PRs (one per platform, probably - packaging for each one comes with its own set of challenges).

I will be more explicit about my requests regarding changes, sorry about that. For now, let's leave these lines as they are, but don't spend too much time on polishing it, I will most likely remove "release" part from matrix and move it to a separate job (as it was originally for MSVC).

Initially I thought we can get away with saving artifacts with release builds only for "on schedule" workflows for master branch (effectively: nightly builds), but the GitHub UI makes it really, really user unfriendly (schedules seem to be half-baked feature) - it would also make it impossible to give WIP builds to testers, so having release builds+artifacts on "push" makes more sense (and is easier to use from GitHub UI).